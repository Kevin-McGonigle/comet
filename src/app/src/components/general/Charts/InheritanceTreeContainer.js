import React from 'react';
import { Pane } from 'evergreen-ui';
import ClassDiagram from '../ClassDiagram/ClassDiagram';
import ForceDirectedGraph from './ForceDirectedGraph';
import { basicRelationshipData } from './configs';

const InheritanceTreeContainer = (props) => {
    const {
        data,
        title,
    } = props;

    return (
        <>
            <ForceDirectedGraph 
                data={basicRelationshipData} 
                graphType="basic"
                title={title}
            />
            <Pane
                display="flex"
                flexWrap="wrap"
                alignItems="center"
                flexGrow={1}
                marginRight="10"
                background="tint2"
                elevation={4}   
                zIndex={-1}
                justifyContent="center"
            >
                <ClassDiagram />
            </Pane>
        </>
    )
}

export default InheritanceTreeContainer;
