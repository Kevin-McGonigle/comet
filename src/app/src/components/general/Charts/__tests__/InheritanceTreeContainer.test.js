import React from 'react';
import InheritanceTreeContainer from '../InheritanceTreeContainer';
import { shallow } from 'enzyme';
import "../../../../setupTests"

test('should render ForceDirectedGraph as expected', () => { 
    const component = shallow(
        <InheritanceTreeContainer title="test" />
    );
    expect(component.html()).toMatchSnapshot();
});
