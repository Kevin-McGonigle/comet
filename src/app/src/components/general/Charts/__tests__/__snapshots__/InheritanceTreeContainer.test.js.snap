// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ForceDirectedGraph as expected 1`] = `"<div class=\\"css-nil ðŸ“¦bs_kymczs ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-wrap_wrap ðŸ“¦flx-grow_1 ðŸ“¦mr_10 ðŸ“¦box-szg_border-box\\"><div class=\\"title\\">test<hr/></div><div id=\\"graph-id-graph-wrapper\\"><svg name=\\"svg-container-graph-id\\" style=\\"height:-200px;width:-800px\\"><defs><marker class=\\"marker\\" id=\\"marker-small\\" viewBox=\\"0 -5 10 10\\" refX=\\"16\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"#2c3e50\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-small-highlighted\\" viewBox=\\"0 -5 10 10\\" refX=\\"16\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"true\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-medium\\" viewBox=\\"0 -5 10 10\\" refX=\\"21.333333333333332\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"#2c3e50\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-medium-highlighted\\" viewBox=\\"0 -5 10 10\\" refX=\\"21.333333333333332\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"true\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-large\\" viewBox=\\"0 -5 10 10\\" refX=\\"26.666666666666664\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"#2c3e50\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker><marker class=\\"marker\\" id=\\"marker-large-highlighted\\" viewBox=\\"0 -5 10 10\\" refX=\\"26.666666666666664\\" refY=\\"0\\" markerWidth=\\"6\\" markerHeight=\\"6\\" orient=\\"auto\\" fill=\\"true\\"><path d=\\"M0,-5L10,0L0,5\\"></path></marker></defs><g id=\\"graph-id-graph-container-zoomable\\" style=\\"transition-duration:0s\\"><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"object,Class A\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"Class A,Class B\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"Class A,Class C\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"Class B,Class D\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"Class D,Class E\\"></path></g><g><path class=\\"link\\" d=\\"M0,0A0,0 0 0,1 0,0\\" style=\\"stroke-width:1.5;stroke:#2c3e50;opacity:1;fill:none;cursor:pointer\\" marker-end=\\"url(#marker-small)\\" id=\\"Class E,Class F\\"></path></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"object\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M-7.0710678118654755,-7.0710678118654755h14.142135623730951v14.142135623730951h-14.142135623730951Z\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"23.5\\" dy=\\".35em\\" fill=\\"#1e272e\\" font-size=\\"20\\" font-weight=\\"normal\\" opacity=\\"1\\">object</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"Class A\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M-7.0710678118654755,-7.0710678118654755h14.142135623730951v14.142135623730951h-14.142135623730951Z\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"23.5\\" dy=\\".35em\\" fill=\\"#1e272e\\" font-size=\\"20\\" font-weight=\\"normal\\" opacity=\\"1\\">Class A</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"Class B\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M-7.0710678118654755,-7.0710678118654755h14.142135623730951v14.142135623730951h-14.142135623730951Z\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"23.5\\" dy=\\".35em\\" fill=\\"#1e272e\\" font-size=\\"20\\" font-weight=\\"normal\\" opacity=\\"1\\">Class B</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"Class C\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M-7.0710678118654755,-7.0710678118654755h14.142135623730951v14.142135623730951h-14.142135623730951Z\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"23.5\\" dy=\\".35em\\" fill=\\"#1e272e\\" font-size=\\"20\\" font-weight=\\"normal\\" opacity=\\"1\\">Class C</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"Class D\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M-7.0710678118654755,-7.0710678118654755h14.142135623730951v14.142135623730951h-14.142135623730951Z\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"23.5\\" dy=\\".35em\\" fill=\\"#1e272e\\" font-size=\\"20\\" font-weight=\\"normal\\" opacity=\\"1\\">Class D</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"Class E\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M-7.0710678118654755,-7.0710678118654755h14.142135623730951v14.142135623730951h-14.142135623730951Z\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"23.5\\" dy=\\".35em\\" fill=\\"#1e272e\\" font-size=\\"20\\" font-weight=\\"normal\\" opacity=\\"1\\">Class E</text></g><g class=\\"node\\" cx=\\"0\\" cy=\\"0\\" id=\\"Class F\\" transform=\\"translate(0,0)\\"><path cursor=\\"pointer\\" opacity=\\"1\\" d=\\"M-7.0710678118654755,-7.0710678118654755h14.142135623730951v14.142135623730951h-14.142135623730951Z\\" fill=\\"#3498db\\" stroke=\\"#3498db\\" stroke-width=\\"1.5\\"></path><text dx=\\"23.5\\" dy=\\".35em\\" fill=\\"#1e272e\\" font-size=\\"20\\" font-weight=\\"normal\\" opacity=\\"1\\">Class F</text></g></g></svg></div></div><div class=\\"css-nil ðŸ“¦bs_kymczs ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-wrap_wrap ðŸ“¦algn-itms_center ðŸ“¦flx-grow_1 ðŸ“¦mr_10 ðŸ“¦z-idx_-1 ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"classDiagramTable\\" id=\\"box\\"><div class=\\"className\\">ClassTest</div><div class=\\"hrLine\\"><hr/></div><div class=\\"classParent\\">Inherits from: ABC</div><div class=\\"hrLine\\"><hr/></div><div class=\\"classArgs\\"><p>+ Arg1 : String</p><p>+ Arg2 : Int</p></div><div class=\\"hrLine\\"><hr/></div><div class=\\"classFunctions\\">+ Function1(Arg1 : String,Arg2 : Int) : String</div></div></div>"`;
